native/core/src/execution/memory_pools/unified_pool.rs

```diff
@@ -76,6 +76,15 @@ impl CometMemoryPool {
     }
 }
 
+impl Drop for CometMemoryPool {
+    fn drop(&mut self) {
+        let used = self.used.load(Relaxed);
+        if used != 0 {
+            log::warn!("CometMemoryPool dropped with {} bytes still reserved", used);
```

Comment:
Would a stacktrace of where this was being dropped be useful? 
Though what we really need is this - [arrow-java HistoricalLog](https://github.com/apache/arrow-java/blob/main/memory/memory-core/src/main/java/org/apache/arrow/memory/util/HistoricalLog.java ). I don't know how to get a combined native+jvm stack trace though.

Reply:
Makes sense